src/CodeGen_OpenCL_Dev.cpp

```diff
@@ -209,8 +211,39 @@ string simt_intrinsic(const string &name) {
 void CodeGen_OpenCL_Dev::CodeGen_OpenCL_C::visit(const For *loop) {
     user_assert(loop->for_type != ForType::GPULane)
         << "The OpenCL backend does not support the gpu_lanes() scheduling directive.";
+       
+    // Cover the added GPUThreadReduceLoop
+    if(loop->for_type == ForType::GPUThreadReduce) {
+        // Check if the condition for the gpu_thread_reduce holds, only for addition (right now)
+        user_assert(loop->body.as<Store>()->value.as<Add>())
+            << "GPUThreadReduce is only defined on addition\n";
+
+        // get the local ID of this tread
+        stream << get_indent() << print_type(Int(32)) << " " << print_name(loop->name)
+               << " = " << simt_intrinsic(loop->name) << ";\n";
 
-    if (is_gpu_var(loop->name)) {
+        // initialize the repsentive sum value (might already be loaded in case of sum[0])
+        stream << get_indent() << "_sum[" << print_name(loop->name) << "] = _sum[" << print_name(loop->name) << "] + "<< print_name(loop->name) << ";\n";
```

Comment:
I think actually you probably want to store the intermediate result in a local array.
Something like `__local int local_sum[1024]`
The 1024 needs to be a constant number, so you can start with just 1024 or whatever, but probably the value is saved in:
`loop->extent`

Furthermore, you hard code `_sum` here.

I think you can get the string of sum like (such that we can name it any other name as well):
```c++
Store* store = loop->body.as<Store>()
string store_name = print_name(store->name);
```

And lastly, instead of `"] + "<< print_name(loop->name) << ";\n";` we can probably write:
```c++
stream << ... << "] + ";
Store* add = loop->body.as<Store>()->value.as<Add>()
add->b.accept() // This will print the contents of b
stream << << ";\n";
```